src/parse_execution.rs

```diff
@@ -1672,6 +1672,8 @@ impl<'a> ParseExecutionContext {
             job.internal_job_id,
         );
 
+        // Start the timer here to ensure command substitution is measured
+        let _timer = push_timer(job.wants_timing());
```

Comment:
Is there a reason you dropped the `no_exec` check here?

Reply:
yes, this function is guarded with a `no_exec` check

https://github.com/fish-shell/fish-shell/blob/14451ff051e099ea6ecd25434a1c5d3f77b65dc0/src/parse_execution.rs#L1564-L1566